content/python/concepts/generators/generators.md

```diff
@@ -0,0 +1,59 @@
+---
+Title: 'Generators'
+Description: 'In Python, a generator is a function or expression that will process a given iterable one object at a time, on demand.'
+Subjects:
+  - 'Computer Science'
+  - 'Data Science'
+Tags:
+  - 'Generators'
+  - 'OOP'
+  - 'Loops'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/data-science'
+---
+
+In Python, a generator is a function or expression that will process a given iterable one object at a time, on demand. A generator will return a generator object, which can be cast as a list or some other data structure as appropriate.
+
+## Generators
+
+Generators are a convenient means of employing [iterator](https://www.codecademy.com/resources/docs/python/iterators) functionality within the syntax of a function or expression. One of the main advantages of generators is that they evaluate items on demand, which means only one item is in memory at a time in lieu of the entire sequence (as with a list).
+
+## A Generator Function
+
+The following code shows the creation of a generator function. In the function definiton the `yield` statement is used to return or include an item in the final generator object.
+
+```codebyte/py
```

Comment:
Can we change this codebyte and the one marked below to a regular example while keeping the codebyte at the end the way it is?
```suggestion
```py
```